### PR TITLE
Check for installed extensions for showing context menus from core in data explorer

### PR DESCRIPTION
**Pull request by abist**
Fixes second part of https://github.com/microsoft/azuredatastudio/issues/7082

### CODE REVIEW

**Comment by anthonydresser** [comment]
Should we only merge this into the release branch? this isn't the right long term fix.

**Comment by abist** [comment]
Sure. Making an extension view contribution point from data explorer would take some work.

**Comment by kburtram** [comment]
I think the correct longer-term fix is to unify the menu item contribution mechanism between Servers and other Connections views.  And having these contributions be provided by the extensions directly.  

I assume this is a medium sized work-item (~1-2 weeks?).  If that's correct I don't know if we'll have time to complete that before Oct release.  So if we take this fix directly to `release/1.11` we'll need to track porting directly to `release/1.12' etc. until the "right" fix is completed.

Given the extension menu implementation details are already in the core I don't think this PR makes the situation significantly worse, as far as I can tell.  I am personally fine with either taking directly to release branch, or merging to master and undoing this change as part of the future refactor.

@abist @anthonydresser do we expect that we'll have time to make the longer-term refactors in this area before Oct release?

File: extensions/agent/package.json [hunk 1]
```diff
@@ -1,93 +1,93 @@
...
-    "mocha-junit-reporter": "^1.17.0",
-    "mocha-multi-reporters": "^1.1.7",
-    "@types/mocha": "^5.2.5",
-    "@types/node": "^8.10.25",
-    "mocha": "^5.2.0",
-    "should": "^13.2.1",
-    "typemoq": "^2.1.0",
-    "vscode": "1.1.5"
-  }
+	"name": "agent",
```
Comment (Charles-Gagnon) [inline extensions/agent/package.json]: It doesn't look like there's any changes in this file asides from spaces -> tabs? Please try to keep unrelated changes like this in separate PRs - it makes it more difficult to figure out what we should actually be reviewing.

File: src/sql/workbench/parts/dataExplorer/common/nodeContext.ts [hunk 2]
```diff
@@ -86,4 +107,36 @@ export class NodeContextKey extends Disposable implements IContextKey<INodeConte
...
+						break;
+					case ('admin-ext-tool-win'):
+						this.adminToolLoaded.set(true);
+						break;
+					default:
+						break;
+				}
+			}
+		}
+		return Promise.resolve();
```
Comment (Charles-Gagnon) [inline src/sql/workbench/parts/dataExplorer/common/nodeContext.ts]: >Promise.resolve(); [](start = 9, length = 18)

You can just return here - no need for the call to Promise.resolve();

File: src/sql/workbench/parts/dataExplorer/common/nodeContext.ts [hunk 3]
```diff
@@ -86,4 +107,36 @@ export class NodeContextKey extends Disposable implements IContextKey<INodeConte
...
 		return this._nodeContextKey.get();
 	}
+
+	/**
+	 * Helper function to get the correct context menu for showing
+	 * extension actions
+	 */
+	public async setExtensionContextKeys(): Promise<void> {
+		for (let extension of inBuiltExtensions) {
+			let extensionLoaded = await this.extensionService.getExtension(extension);
```
Comment (Charles-Gagnon) [inline src/sql/workbench/parts/dataExplorer/common/nodeContext.ts]: What about if a user installs/uninstalls an extension? We don't reload ADS for the majority of them anymore so the menu items I wouldn't expect to show up in that case which isn't very desireable.

**Comment by anthonydresser** [comment]
The long long term fix might be to merge the two extension points but the short long term fix is to just make contributing to dataExplorer/context from an extension work. Currently the issue is you don't get any meaningful info when you contribute to that menu point. This work shouldn't be too much work, I already have a good idea what the implementation would look like.

**Comment by abist** [comment]
Closing this since we don't want it in September and it's not a regression.
